src/super_gradients/training/utils/quantization/core.py

```diff
@@ -0,0 +1,140 @@
+import inspect
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union, Type, Optional
+
+from pytorch_quantization.nn.modules._utils import QuantMixin, QuantInputMixin
+from pytorch_quantization.tensor_quant import QuantDescriptor
+from torch import nn
+
+
+def _extract_init_args(cls, float_instance):
+    """
+    Inspecting the __init__ args, and searching for corresponding properties from the float instance
+    e.g., for `__init__(self, a)` the mechanism will look for `float_instance.a` and pass that value to `__init__`
+    """
+    required_init_params = list(inspect.signature(cls.__init__).parameters)[1:]  # [0] is self
+
+    if 'kwargs' in required_init_params:  # we don't want to search for a state named `kwargs`
+        required_init_params.pop(required_init_params.index('kwargs'))
+
+    float_instance_state = {}
+    for p in required_init_params:
+        if not hasattr(float_instance, p):
+            raise ValueError(f"{float_instance.__class__.__name__} is missing `{p}` which is required "
+                             f"in {cls.__name__}.__init__. Either override `SGQuantBase.from_float` "
+                             f"or add {p} as state for {float_instance.__class__.__name__}.")
+        float_instance_state[p] = getattr(float_instance, p)
+
+    # Edge-cases
+    if 'bias' in float_instance_state:
+        if float_instance_state['bias'] is None:  # None is the state when bias=False in torch.nn
+            float_instance_state['bias'] = False
+        elif not isinstance(float_instance_state['bias'], bool):  # Tensor is the state when bias=True in torch.nn
+            float_instance_state['bias'] = True
+        # in case bias is a boolean - we don't do anything, so it is taken as-is, either True or False
+    return float_instance_state
+
+
+def _from_float(cls, float_instance, **kwargs):
```

Comment:
If my question in my first comment is true, then we need to allow passing _extract_init_args to this function as well.

Reply:
To do custom stuff, one may override `SGQuantMixin.from_float`